biom/parse.py

```diff
@@ -245,6 +246,98 @@ def get_axis_indices(biom_str, to_keep, axis):
     return idxs, json.dumps(subset)[1:-1]  # trim off { and }
 
 
+def parse_uc(fh):
+    """ Create a Table object from a uclust/usearch/vsearch uc file.
+
+        Parameters
+        ----------
+        fh : file handle
+            The ``.uc`` file to be parsed.
+
+        Returns
+        -------
+        biom.Table : The resulting BIOM table.
+
+        Raises
+        ------
+        ValueError
+            If a sequence identifier is encountered that doesn't have at least
+            one underscore in it (see Notes).
+
+        Notes
+        -----
+        This file assumes that sequence identifiers in this file are in QIIME's
+        "post-split-libraries" format, where the identifiers are of the form
+        ``<sample-id>_<sequence-id>``. Everything before the first underscore
+        will be used as the sample identifier in the resulting ``Table``.
+        The information after the first underscore is not used directly, though
+        the full identifiers of seeds will be used as the observation
+        identifier in the resulting ``Table``.
+
+    """
+    data = collections.defaultdict(int)
+    sample_idxs = {}
+    sample_ids = []
+    observation_idxs = {}
+    observation_ids = []
+    # The types of hit lines we need here are hit (H), seed (S) and
+    # library seed (L). Store these in a set for quick reference.
+    line_types = set(list('HSL'))
+    for line in fh:
+        # determine if the current line is one that we need
+        line = line.strip()
+        if not (line and line[0] in line_types):
```

Comment:
kk

On Mon, Oct 19, 2015 at 2:12 PM, Greg Caporaso <notifications@github.com>
wrote:

> In biom/parse.py
> <https://github.com/biocore/biom-format/pull/657#discussion_r42419690>:
>
> > +        the full identifiers of seeds will be used as the observation
> > +        identifier in the resulting ``Table``.
> > +
> > +    """
> > +    data = collections.defaultdict(int)
> > +    sample_idxs = {}
> > +    sample_ids = []
> > +    observation_idxs = {}
> > +    observation_ids = []
> > +    # The types of hit lines we need here are hit (H), seed (S) and
> > +    # library seed (L). Store these in a set for quick reference.
> > +    line_types = set(list('HSL'))
> > +    for line in fh:
> > +        # determine if the current line is one that we need
> > +        line = line.strip()
> > +        if not (line and line[0] in line_types):
>
> Eh, I'd rather not adapt this for regex now. I'd prefer to just leave as
> it since it works.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/biom-format/pull/657/files#r42419690>.
>